cmd/gateway/zcn/dStorage.go

```diff
@@ -77,43 +77,50 @@ func listRegularRefs(alloc *sdk.Allocation, remotePath, marker, fileType string,
 	var markedPath string
 
 	remotePath = filepath.Clean(remotePath)
-	commonPrefix := getCommonPrefix(remotePath)
-	offsetPath := filepath.Join(remotePath, marker)
-	for {
-		oResult, err := getRegularRefs(alloc, remotePath, offsetPath, fileType, pageLimit)
-		if err != nil {
-			return nil, true, "", nil, err
-		}
-		if len(oResult.Refs) == 0 {
-			break
-		}
 
-		for i := 0; i < len(oResult.Refs); i++ {
-			ref := oResult.Refs[i]
-			trimmedPath := strings.TrimPrefix(ref.Path, remotePath+"/")
-			if isDelimited {
+	directories := []string{remotePath}
```

Comment:
why do we need a loop if there is only one element in the slice?

Reply:
Initially there is only one element but for the recursive flags i.e `ls -r` we need to find the files inside the nested directories and in those cases this directories slice would be appended. hence the loop.